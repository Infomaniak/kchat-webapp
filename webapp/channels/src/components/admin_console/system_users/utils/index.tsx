// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {SortingState} from '@tanstack/react-table';
import React from 'react';
import {FormattedMessage} from 'react-intl';

import {UserReportSortColumns, ReportSortDirection} from '@mattermost/types/reports';
import type {UserReportOptions, UserReport} from '@mattermost/types/reports';
import type {Team} from '@mattermost/types/teams';

import {PAGE_SIZES} from 'components/admin_console/list_table';

import type {AdminConsoleUserManagementTableProperties} from 'types/store/views';

import {ColumnNames, RoleFilters, StatusFilter, TeamFilters} from '../constants';
import type {TableOptions} from '../system_users';
import type {OptionType as TeamFilterOptionType} from '../system_users_filters_popover/system_users_filter_team';

export function convertTableOptionsToUserReportOptions(tableOptions?: TableOptions): UserReportOptions {
    return {
        page_size: tableOptions?.pageSize || PAGE_SIZES[0],
        from_column_value: tableOptions?.fromColumnValue,
        from_id: tableOptions?.fromId,
        direction: tableOptions?.direction,
        ...getSortColumnForOptions(tableOptions?.sortColumn),
        ...getSortDirectionForOptions(tableOptions?.sortIsDescending),
        ...getSearchFilterOption(tableOptions?.searchTerm),
        ...getTeamFilterOption(tableOptions?.filterTeam),
        ...getStatusFilterOption(tableOptions?.filterStatus),
        ...getRoleFilterOption(tableOptions?.filterRole),
        date_range: tableOptions?.dateRange,
    };
}

/**
 * Converts the sorting column name to API compatible sorting column name. Default sorting column name is by username.
 */
export function getSortColumnForOptions(id?: SortingState[0]['id']): Pick<UserReportOptions, 'sort_column'> {
    let sortColumn: UserReportOptions['sort_column'];

    if (id === ColumnNames.email) {
        sortColumn = UserReportSortColumns.email;
    } else if (id === ColumnNames.createAt) {
        sortColumn = UserReportSortColumns.createAt;
    } else {
        // Default sorting to first User details column
        sortColumn = UserReportSortColumns.username;
    }

    return {
        sort_column: sortColumn,
    };
}

/**
 * Converts the sorting direction to API compatible sorting direction. Default sorting direction is ascending.
 */
export function getSortDirectionForOptions(desc?: SortingState[0]['desc']): Pick<UserReportOptions, 'sort_direction'> {
    let sortDirection: UserReportOptions['sort_direction'];

    if (desc) {
        sortDirection = ReportSortDirection.descending;
    } else {
        sortDirection = ReportSortDirection.ascending;
    }

    return {
        sort_direction: sortDirection,
    };
}

/**
 * It returns the value of that column on which sorting is applied.
 */
export function getSortableColumnValueBySortColumn(row: UserReport, sortColumn: AdminConsoleUserManagementTableProperties['sortColumn']): string {
    switch (sortColumn) {
    case ColumnNames.email:
        return row.email;
    case ColumnNames.createAt:
        return String(row.create_at);
    default:
        return row.username;
    }
}

export function getTeamFilterOption(teamId?: string): Partial<Pick<UserReportOptions, 'team_filter' | 'has_no_team'>> {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            team_filter: undefined,
            has_no_team: undefined,
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            team_filter: undefined,
            has_no_team: true,
        };
    }

    return {
        team_filter: teamId,
        has_no_team: undefined,
    };
}

export function getStatusFilterOption(status?: string): Partial<Pick<UserReportOptions, 'hide_active' | 'hide_inactive'>> {
    if (status === StatusFilter.Active) {
        return {
            hide_inactive: true,
        };
    } else if (status === StatusFilter.Deactivated) {
        return {
            hide_active: true,
        };
    }

    return {
        hide_active: undefined,
        hide_inactive: undefined,
    };
}

export function getPaginationInfo(pageIndex: number, pageSize: number, currentLength: number, total?: number) {
    if (!currentLength) {
        return (
            <FormattedMessage
                id='admin.system_users_list.pagination.no_users'
                defaultMessage='0 users'
            />
        );
    }

    const firstPage = (pageIndex * pageSize) + 1;
    const lastPage = (pageIndex * pageSize) + currentLength;
    const totalItems = total || 0;

    return (
        <FormattedMessage
            id='admin.system_users_list.pagination'
            defaultMessage='Showing {firstPage} - {lastPage} of {totalItems} users'
            values={{
                firstPage,
                lastPage,
                totalItems,
            }}
        />
    );
}

export function getDefaultSelectedValueFromList<T extends {value: string}>(value: string, options: T[]) {
    const option = options.find((option) => option.value === value);

    if (option) {
        return option;
    }

    return options[0];
}

export function getDefaultSelectedTeam(teamId: Team['id'] | string, label?: string): TeamFilterOptionType {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            value: TeamFilters.AllTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.allTeams'
                    defaultMessage='All Teams'
                />
            ),
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            value: TeamFilters.NoTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.noTeams'
                    defaultMessage='No Teams'
                />
            ),
        };
    }

    return {
        value: teamId,
        label: label || '',
    };
}

export function getRoleFilterOption(role?: string): Pick<UserReportOptions, 'role_filter'> {
    if (!role || role === RoleFilters.Any) {
        return {role_filter: undefined};
    }
    return {role_filter: role};
}

export function getSearchFilterOption(search?: string): Pick<UserReportOptions, 'search_term'> {
    if (!search || search.trim().length === 0) {
        return {search_term: undefined};
    }

    return {search_term: search};
}
