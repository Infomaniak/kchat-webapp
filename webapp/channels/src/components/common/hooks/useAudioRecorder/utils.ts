export const getWasmFileURL = () =>
    new URL('wasm-media-encoders/wasm/mp3', import.meta.url);
