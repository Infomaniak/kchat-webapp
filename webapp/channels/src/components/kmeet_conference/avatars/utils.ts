export function countMeta<T>(
    items: T[],
    total = items.length,
): [T[], T[], {overflowUnnamedCount: number; nonDisplayCount: number}] {
    const breakAt = Math.max(items.length, total) > 4 ? 3 : 4;

    const displayItems = items.slice(0, breakAt);
    const overflowItems = items.slice(breakAt);

    const overflowUnnamedCount = Math.max(total - displayItems.length - overflowItems.length, 0);
    const nonDisplayCount = overflowItems.length + overflowUnnamedCount;

    return [displayItems, overflowItems, {overflowUnnamedCount, nonDisplayCount}];
}
