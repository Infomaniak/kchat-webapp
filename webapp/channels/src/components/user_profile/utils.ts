// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import ColorContrastChecker from 'color-contrast-checker';
import ColorHash from 'color-hash';

const REQUIRED_COLOR_RATIO = 4.5;

export const cachedUserNameColors = new Map<string, string>();

export function generateColor(username: string, background: string): string {
    const cacheKey = `${username}-${background}`;
    const cachedColor = cachedUserNameColors.get(cacheKey);
    if (cachedColor) {
        return cachedColor;
    }

    let userColor = background;
    let contrastRatio = 1;
    let userAndSalt = username;
    const checker = new ColorContrastChecker();
    const colorHash = new ColorHash();

    const backgroundLuminance = checker.hexToLuminance(background);
    for (let tries = 10; tries > 0; tries--) {
        const textColor = colorHash.hex(userAndSalt);

        const cr = checker.getContrastRatio(
            checker.hexToLuminance(textColor),
            backgroundLuminance,
        );

        if (cr > contrastRatio) {
            userColor = textColor;
            contrastRatio = cr;
        }

        if (cr >= REQUIRED_COLOR_RATIO) {
            break;
        }

        userAndSalt += 'salt';
    }

    cachedUserNameColors.set(cacheKey, userColor);
    return userColor;
}
