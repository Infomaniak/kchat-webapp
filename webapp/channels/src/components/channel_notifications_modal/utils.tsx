// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage, defineMessages} from 'react-intl';

import type {ChannelNotifyProps} from '@mattermost/types/channels';
import type {UserNotifyProps} from '@mattermost/types/users';

import type {FieldsetCheckbox} from 'components/widgets/modals/components/checkbox_setting_item';
import type {FieldsetRadio} from 'components/widgets/modals/components/radio_setting_item';

import {NotificationLevels} from 'utils/constants';

export type ChannelMemberNotifyProps = Partial<ChannelNotifyProps> & Pick<UserNotifyProps, 'desktop_threads' | 'push_threads'>

const translations = defineMessages({
    MuteAndIgnoreSectionTitle: {
        id: 'channel_notifications.muteAndIgnore',
        defaultMessage: 'Mute or ignore',
    },
    NotifyMeTitle: {
        id: 'channel_notifications.NotifyMeTitle',
        defaultMessage: 'Notify me about…',
    },
    ThreadsReplyTitle: {
        id: 'channel_notifications.ThreadsReplyTitle',
        defaultMessage: 'Thread reply notifications',
    },

    DesktopNotificationsSectionTitle: {
        id: 'channel_notifications.desktopNotificationsTitle',
        defaultMessage: 'Desktop Notifications',
    },

    DesktopNotificationsSectionDesc: {
        id: 'channel_notifications.desktopNotificationsDesc',
        defaultMessage: 'Available on Chrome, Edge, Firefox, and the Mattermost Desktop App.',
    },

    MobileNotificationsSectionTitle: {
        id: 'channel_notifications.mobileNotificationsTitle',
        defaultMessage: 'Mobile Notifications',
    },

    MobileNotificationsSectionDesc: {
        id: 'channel_notifications.mobileNotificationsDesc',
        defaultMessage: 'Notification alerts are pushed to your mobile device when there is activity in Mattermost.',
    },

    MuteChannelDesc: {
        id: 'channel_notifications.muteChannelDesc',
        defaultMessage: 'Turns off notifications for this channel. You’ll still see badges if you’re mentioned.',
    },

    IgnoreMentionsDesc: {
        id: 'channel_notifications.ignoreMentionsDesc',
        defaultMessage: 'When enabled, @channel, @here and @all will not trigger mentions or mention notifications in this channel',
    },

    MuteChannelInputFieldTitle: {
        id: 'channel_notifications.muteChannelTitle',
        defaultMessage: 'Mute channel',
    },

    DesktopReplyThreadsInputFieldTitle: {
        id: 'channel_notifications.checkbox.threadsReplyTitle',
        defaultMessage: 'Notify me about replies to threads I\'m following',
    },

    MobileReplyThreadsInputFieldTitle: {
        id: 'channel_notifications.checkbox.threadsReplyTitle',
        defaultMessage: 'Notify me about replies to threads I\'m following',
    },

    sameMobileSettingsDesktopInputFieldTitle: {
        id: 'channel_notifications.checkbox.sameMobileSettingsDesktop',
        defaultMessage: 'Use the same notification settings as desktop',
    },

    IgnoreMentionsInputFieldTitle: {
        id: 'channel_notifications.ignoreMentionsTitle',
        defaultMessage: 'Ignore mentions for @channel, @here and @all',
    },

    AutoFollowThreadsTitle: {
        id: 'channel_notifications.autoFollowThreadsTitle',
        defaultMessage: 'Follow all threads in this channel',
    },

    AutoFollowThreadsDesc: {
        id: 'channel_notifications.autoFollowThreadsDesc',
        defaultMessage: 'When enabled, all new replies in this channel will be automatically followed and will appear in your Threads view.',
    },

    AutoFollowThreadsInputFieldTitle: {
        id: 'channel_notifications.checkbox.autoFollowThreadsTitle',
        defaultMessage: 'Automatically follow threads in this channel',
    },
});

const desktopNotificationInputFieldOptions = defineMessages({
    allNewMessages: {
        id: 'channel_notifications.desktopNotificationAllLabel',
        defaultMessage: 'All new messages',
    },
    mentions: {
        id: 'channel_notifications.desktopNotificationMentionLabel',
        defaultMessage: 'Mentions, direct messages, and keywords only',
    },
    nothing: {
        id: 'channel_notifications.desktopNotificationNothingLabel',
        defaultMessage: 'Nothing',
    },
});

const mobileNotificationInputFieldOptions = defineMessages({
    allNewMessages: {
        id: 'channel_notifications.MobileNotificationAllLabel',
        defaultMessage: 'All new messages',
    },
    mentions: {
        id: 'channel_notifications.MobileNotificationMentionLabel',
        defaultMessage: 'Mentions, direct messages, and keywords only',
    },
    nothing: {
        id: 'channel_notifications.MobileNotificationNothingLabel',
        defaultMessage: 'Nothing',
    },
});

const MuteChannelInputFieldData: FieldsetCheckbox = {
    title: translations.MuteChannelInputFieldTitle,
    name: 'mute channel',
    dataTestId: 'muteChannel',
};

const DesktopReplyThreadsInputFieldData: FieldsetCheckbox = {
    title: translations.DesktopReplyThreadsInputFieldTitle,
    name: 'desktop reply threads',
    dataTestId: 'desktopReplyThreads',
};

const MobileReplyThreadsInputFieldData: FieldsetCheckbox = {
    title: translations.MobileReplyThreadsInputFieldTitle,
    name: 'mobile reply threads',
    dataTestId: 'mobileReplyThreads',
};

const sameMobileSettingsDesktopInputFieldData: FieldsetCheckbox = {
    title: translations.sameMobileSettingsDesktopInputFieldTitle,
    name: 'same mobile settings as Desktop',
    dataTestId: 'sameMobileSettingsDesktop',
};

const IgnoreMentionsInputFieldData: FieldsetCheckbox = {
    title: translations.IgnoreMentionsInputFieldTitle,
    name: 'ignore mentions',
    dataTestId: 'ignoreMentions',
};

const AutoFollowThreadsInputFieldData: FieldsetCheckbox = {
    title: translations.AutoFollowThreadsInputFieldTitle,
    name: 'auto follow threads',
    dataTestId: 'autoFollowThreads',
};

const desktopNotificationInputFieldData = (defaultOption: string): FieldsetRadio => {
    return {
        options: [
            {
                dataTestId: `desktopNotification-${NotificationLevels.ALL}`,
                title: desktopNotificationInputFieldOptions.allNewMessages,
                name: `desktopNotification-${NotificationLevels.ALL}`,
                key: `desktopNotification-${NotificationLevels.ALL}`,
                value: NotificationLevels.ALL,
                suffix: defaultOption === NotificationLevels.ALL ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `desktopNotification-${NotificationLevels.MENTION}`,
                title: desktopNotificationInputFieldOptions.mentions,
                name: `desktopNotification-${NotificationLevels.MENTION}`,
                key: `desktopNotification-${NotificationLevels.MENTION}`,
                value: NotificationLevels.MENTION,
                suffix: defaultOption === NotificationLevels.MENTION ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `desktopNotification-${NotificationLevels.NONE}`,
                title: desktopNotificationInputFieldOptions.nothing,
                name: `desktopNotification-${NotificationLevels.NONE}`,
                key: `desktopNotification-${NotificationLevels.NONE}`,
                value: NotificationLevels.NONE,
                suffix: defaultOption === NotificationLevels.NONE ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
        ],
    };
};

const mobileNotificationInputFieldData = (defaultOption: string): FieldsetRadio => {
    return {
        options: [
            {
                dataTestId: `MobileNotification-${NotificationLevels.ALL}`,
                title: mobileNotificationInputFieldOptions.allNewMessages,
                name: `MobileNotification-${NotificationLevels.ALL}`,
                key: `MobileNotification-${NotificationLevels.ALL}`,
                value: NotificationLevels.ALL,
                suffix: defaultOption === NotificationLevels.ALL ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `MobileNotification-${NotificationLevels.MENTION}`,
                title: mobileNotificationInputFieldOptions.mentions,
                name: `MobileNotification-${NotificationLevels.MENTION}`,
                key: `MobileNotification-${NotificationLevels.MENTION}`,
                value: NotificationLevels.MENTION,
                suffix: defaultOption === NotificationLevels.MENTION ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `MobileNotification-${NotificationLevels.NONE}`,
                title: mobileNotificationInputFieldOptions.nothing,
                name: `MobileNotification-${NotificationLevels.NONE}`,
                key: `MobileNotification-${NotificationLevels.NONE}`,
                value: NotificationLevels.NONE,
                suffix: defaultOption === NotificationLevels.NONE ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
        ],
    };
};

const utils = {
    desktopNotificationInputFieldData,
    DesktopNotificationsSectionDesc: translations.DesktopNotificationsSectionDesc,
    DesktopNotificationsSectionTitle: translations.DesktopNotificationsSectionTitle,
    IgnoreMentionsDesc: translations.IgnoreMentionsDesc,
    IgnoreMentionsInputFieldData,
    mobileNotificationInputFieldData,
    MobileNotificationsSectionDesc: translations.MobileNotificationsSectionDesc,
    MobileNotificationsSectionTitle: translations.MobileNotificationsSectionTitle,
    MuteAndIgnoreSectionTitle: translations.MuteAndIgnoreSectionTitle,
    MuteChannelDesc: translations.MuteChannelDesc,
    MuteChannelInputFieldData,
    NotifyMeTitle: translations.NotifyMeTitle,
    DesktopReplyThreadsInputFieldData,
    ThreadsReplyTitle: translations.ThreadsReplyTitle,
    MobileReplyThreadsInputFieldData,
    AutoFollowThreadsTitle: translations.AutoFollowThreadsTitle,
    AutoFollowThreadsDesc: translations.AutoFollowThreadsDesc,
    AutoFollowThreadsInputFieldData,
    sameMobileSettingsDesktopInputFieldData,
};

export default utils;
