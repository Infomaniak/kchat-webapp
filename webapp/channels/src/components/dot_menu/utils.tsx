// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';
import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

import {trackEvent} from 'actions/telemetry_actions';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
