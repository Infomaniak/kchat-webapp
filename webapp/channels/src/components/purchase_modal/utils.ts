// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {Product} from '@mattermost/types/cloud';

import {CloudProducts, RecurringIntervals} from 'utils/constants';

/**
 *
 * @param products  Record<string, Product> | undefined - the list of current cloud products
 * @param productId String - a valid product id used to find a particular product in the dictionary
 * @param productSku String - the sku value of the product of type either cloud-starter | cloud-professional | cloud-enterprise
 * @returns Product
 */
export function findProductInDictionary(products: Record<string, Product> | undefined, productId?: string | null, productSku?: string, productRecurringInterval?: string): Product | null {
    if (!products) {
        return null;
    }
    const keys = Object.keys(products);
    if (!keys.length) {
        return null;
    }
    if (!productId && !productSku) {
        return products[keys[0]];
    }
    let currentProduct = products[keys[0]];
    if (keys.length > 1) {
        // here find the product by the provided id or name, otherwise return the one with Professional in the name
        keys.forEach((key) => {
            if (productId && products[key].id === productId) {
                currentProduct = products[key];
            } else if (productSku && products[key].sku === productSku && products[key].recurring_interval === productRecurringInterval) {
                currentProduct = products[key];
            }
        });
    }
    return currentProduct;
}

export function getSelectedProduct(yearlyProducts: Record<string, Product>) {
    return findProductInDictionary(yearlyProducts, null, CloudProducts.PROFESSIONAL, RecurringIntervals.YEAR);
}
