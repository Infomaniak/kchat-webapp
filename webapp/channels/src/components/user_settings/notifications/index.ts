// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect, type ConnectedProps} from 'react-redux';

import {updateMe} from 'mattermost-redux/actions/users';
import {getSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
import {isCollapsedThreadsEnabled} from 'mattermost-redux/selectors/entities/preferences';

import {isCallsEnabled, isCallsRingingEnabledOnServer} from 'selectors/calls';

import {isEnterpriseOrCloudOrSKUStarterFree} from 'utils/license_utils';

import type {GlobalState} from 'types/store';

import UserSettingsNotifications from './user_settings_notifications';

const mapStateToProps = (state: GlobalState) => {
    const config = getConfig(state);

    const sendPushNotifications = config.SendPushNotifications === 'true';
    const enableAutoResponder = config.ExperimentalEnableAutomaticReplies === 'true';

    const license = getLicense(state);
    const subscriptionProduct = getSubscriptionProduct(state);

    const isEnterpriseReady = config.BuildEnterpriseReady === 'true';

    return {
        sendPushNotifications,
        enableAutoResponder,
        isCollapsedThreadsEnabled: isCollapsedThreadsEnabled(state),
        isCallsRingingEnabled: isCallsEnabled(state, '0.17.0') && isCallsRingingEnabledOnServer(state),
        isEnterpriseOrCloudOrSKUStarterFree: isEnterpriseOrCloudOrSKUStarterFree(license, subscriptionProduct, isEnterpriseReady),
        isEnterpriseReady,

    };
};

const mapDispatchToProps = {
    updateMe,
};

const connector = connect(mapStateToProps, mapDispatchToProps);

export type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(UserSettingsNotifications);
