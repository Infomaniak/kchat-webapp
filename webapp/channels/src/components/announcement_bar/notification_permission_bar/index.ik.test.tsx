// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import * as utilsNotifications from 'utils/notifications';
import {isMobileBrowser} from 'utils/user_agent';

import {renderWithContext, screen} from 'tests/react_testing_utils';

import NotificationPermissionBar from './index';

jest.mock('utils/user_agent', () => ({
    ...jest.requireActual('utils/user_agent'),
    isMobileBrowser: jest.fn(),
}));

describe('NotificationPermissionBar', () => {
    const initialState = {
        entities: {
            users: {
                currentUserId: 'user-id',
            },
            general: {
                config: {},
                license: {},
            },
        },
    };

    beforeEach(() => {
        jest.spyOn(utilsNotifications, 'isNotificationAPISupported').mockReturnValue(false);
        (isMobileBrowser as jest.Mock).mockReturnValue(false);
    });

    afterEach(() => {
        jest.restoreAllMocks();
    });

    test('should render the NotificationUnsupportedBar if notifications are not supported and NOT on mobile', () => {
        const {container} = renderWithContext(<NotificationPermissionBar/>, initialState);

        expect(container).toMatchSnapshot();

        expect(screen.queryByText('Your browser does not support browser notifications.')).toBeInTheDocument();
        expect(screen.queryByText('Update your browser')).toBeInTheDocument();
    });

    test('should NOT render the NotificationUnsupportedBar if notifications are not supported on mobile', () => {
        (isMobileBrowser as jest.Mock).mockReturnValue(true);

        const {container} = renderWithContext(<NotificationPermissionBar/>, initialState);

        expect(container).toMatchSnapshot();

        expect(screen.queryByText('Your browser does not support browser notifications.')).not.toBeInTheDocument();
        expect(screen.queryByText('Update your browser')).not.toBeInTheDocument();
    });
});
