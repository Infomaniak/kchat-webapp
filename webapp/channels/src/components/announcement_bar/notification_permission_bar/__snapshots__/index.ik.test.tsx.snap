// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationPermissionBar should NOT render the NotificationUnsupportedBar if notifications are not supported on mobile 1`] = `<div />`;

exports[`NotificationPermissionBar should render the NotificationUnsupportedBar if notifications are not supported and NOT on mobile 1`] = `
<div>
  <div
    class="_StyledDiv-BRlth ksRYxX announcement-bar"
  >
    <div
      class="announcement-bar__text"
    >
      <i
        class="icon icon-alert-circle-outline"
      />
      <span>
        Your browser does not support browser notifications.
      </span>
      <button>
        Update your browser
      </button>
    </div>
    <a
      class="announcement-bar__close"
      href="#"
    >
      Ã—
    </a>
  </div>
</div>
`;
